scripts/fast-reboot

```diff
@@ -483,8 +519,16 @@ systemctl stop swss
 if [[ "$REBOOT_TYPE" = "warm-reboot" || "$REBOOT_TYPE" = "fastfast-reboot" ]]; then
     initialize_pre_shutdown
 
+    if [[ "x$sonic_asic_type" == x"mellanox" ]]; then
+        check_issu_bank_file
+    fi
+
     request_pre_shutdown
 
+    if [[ "x$sonic_asic_type" == x"mellanox" ]]; then
+        check_issu_bank_file
+    fi
```

Comment:
Should the 2nd check after `wait_for_pre_shutdown_complete_or_fail`?

Reply:
This is kind of too late.

The check happens past the point of no return and the warm reboot will happen regardless. Can this check happen with rest of the pre-checks?

Reply:
To my knowledge, issu_bank.txt is written on ISSU start == request_pre_shutdown. There is actually no point to check it before - it can even not exist.  The corruption may happen only after ISSU start, but since it is no return point then this check is useless and I don't see a good point to place a check like this. If the file is corrupted before ISSU start it is ok, ISSU start will write the correct value to the file. If the file is corrupted after ISSU start we can't really perform a check since this is too late. Now we can either proceed and rely on the fix in recent SDK or abort warm reboot at the stage when cold reboot is required to restore the system as done in this PR.

Reply:
@stepanblyschak Could you explain "ISSU start == request_pre_shutdown" in details?

Let's assume the file is already corrupted in a running switch, and we run `sudo warm-reboot`. How should precheck to protect?

Reply:
request_pre_shutdown -> syncd sets PRESHUTDOWN attribute, mellanox SAI calls ISSU start -> issu start writes to /host/warmboot/issu_bank.txt.

The precheck could be only done if you want to abort warm reboot. However, it will result in a broken switch that anyway will require cold reboot (not just fixing /host/warmboot/issu_bank.txt).

Reply:
My question is still applicable.

---
In reply to: [427514083](https://github.com/Azure/sonic-utilities/pull/915#discussion_r427514083) [](ancestors = 427514083)

Reply:
Offline discussed with Ying, this one is resolved.

---
In reply to: [427688717](https://github.com/Azure/sonic-utilities/pull/915#discussion_r427688717) [](ancestors = 427688717)

Reply:
> My question is still applicable.
> 
> In reply to: [427514083](https://github.com/Azure/sonic-utilities/pull/915#discussion_r427514083) [](ancestors = 427514083)

Fixed